numpy/lib/npyio.py

```diff
@@ -1006,7 +1020,18 @@ def split_line(line):
                 except ValueError:
                     # Unused converter specified
                     continue
-            converters[i] = conv
+            if byte_converters:
+                # converters may use decode to workaround numpy's oldd behaviour,
+                # so encode the string again before passing to the user converter
+                def tobytes_first(x, conv):
+                    if type(x) == bytes:
+                        return conv(x)
+                    return conv(x.encode("latin1"))
+                import functools
+                user_conv = functools.partial(tobytes_first, conv=conv)
+                converters[i] = user_conv
+            else:
+                converters[i] = conv
```

Comment:
Maybe here put the check on `conv is bytes` from below? It seems nicer to just ensure the converters are all set up correctly at this stage.